### ISSUE TITLE
Inserted images have incorrect colors in dark mode

### ISSUE THREAD

**Issue by karlhorky** [excalidraw/excalidraw#6516]
When switching Excalidraw to dark mode, the color representation of images is off (tested browser: Chrome 112 on macOS):

https://user-images.githubusercontent.com/1935696/234062899-3e4c69d8-1f9a-40d6-93cd-87df9fdf2ddc.mp4

Original screenshot:

<img width="1440" alt="Screenshot 2023-04-19 at 15 07 35" src="https://user-images.githubusercontent.com/1935696/234062961-09149c41-7abd-4edf-b9ec-b672252c16d7.png">

This color shift issue (with dark mode) also has some explanation over here (although the author @anatoliibakhtin is actually reporting a different issue):

- https://github.com/excalidraw/excalidraw/issues/5312

It is apparently caused by the color inversion:

> @dwelle in [comment 1157697888](https://github.com/excalidraw/excalidraw/issues/5312#issuecomment-1157697888): we invert the colors using CSS, and then individually apply a counter filter on image elements

Also mentioned in #4616:

> @dwelle in [comment ](https://github.com/excalidraw/excalidraw/issues/4616#issuecomment-1250112100): Not something we do by choice, but due to how our dark mode works right now.
> 
> I assume you're using Safari? On other browsers that support canvas filters we selectively invert the images back (the double invert isn't 100% accurate, but for now it seems serviceable).

I am in Chrome 112 macOS, and this is a noticeable, undesirable color shift.

Would be amazing to improve the "counter filter" to work better in non-Safari browsers (also in Safari would be ideal)

**Comment by dwelle** [comment]
Yeah, the double filter is not ideal and also canvas filters don't work on Safari as mentioned. The solution is to move away from using filters for dark mode, and invert the colors programmatically in JS.

**Comment by beyonxd** [comment]
I would like to work on this issue, also I want some suggestions like you want me to remove the filter and implement the color inversion using js or something else? 
Do I need to get this issue assigned first or simply making a PR would be good ?

**Comment by rePewtr** [comment]
would love to see this, using excalidraw for creative work where color accuarcy in photos is crucial. Dark mode is basically unusable for that now. And I do love me some dark mode :)

**Comment by petitpandarouge** [comment]
I don't know if it's correlated, but the Windows emojy suffers the same issue.
Maybe it's different from the images as the emojy are considered as text, I don't know...

![image](https://github.com/excalidraw/excalidraw/assets/15639809/ef9c9241-d38a-489c-9d3d-9d1b3bd42dc3)

**Comment by DudeThatsErin** [comment]
I'm seeing this on iPad via the Obsidian Plugin.

reported via discord server a week ago and no response but I asked again about it there.
![image](https://github.com/user-attachments/assets/c7c3e017-5c74-4e73-b892-ea4577a6ae7f)
![image](https://github.com/user-attachments/assets/55b074f6-6a67-41cb-a207-d04ab3697139)

Browser picture is how it is supposed to look and how it does look on my desktop. Second screenshot is from the plugin on my iPad.

**Comment by avilabiel** [comment]
I just got this issue. It's funny, but pls fix it haha

**Comment by LaurieAnnis** [comment]
> The solution is to move away from using filters for dark mode, and invert the colors programmatically in JS.

I second this.  I am using Chrome on Windows.  I need my pasted images to remain their correct colors, while retaining a dark background and light color palette for drawn and text objects, otherwise I will need to use Microsoft Whiteboard (the horror!)  

Right now my workaround is drawing a large black box in light mode, locking it, editing all my drawing and text colors to white/light, and then screen capturing the results.  Definitely not optimal for a productivity tool.

**Comment by Mohamed3nan** [comment]
"Not only are images affected, but colours also appear differently in dark and light modes!!

![Image](https://github.com/user-attachments/assets/2eb50c48-d5a0-43b5-add4-725e49d14c84)

![Image](https://github.com/user-attachments/assets/71e4c315-d1ef-4d1d-885a-0e7c9d30abed)

**Comment by dwelle** [comment]
fixed by https://github.com/excalidraw/excalidraw/pull/10578
